Sources/MongoSwift/BSON/AnyBSONValue.swift

```diff
@@ -58,6 +58,15 @@ public struct AnyBSONValue: Codable, Equatable {
     public init(from decoder: Decoder) throws {
         // short-circuit in the `BSONDecoder` case
         if let bsonDecoder = decoder as? _BSONDecoder {
+            if bsonDecoder.storage.topContainer as? Date != nil {
+                switch bsonDecoder.options.dateDecodingStrategy {
+                case BSONDecoder.DateDecodingStrategy.bsonDateTime:
```

Comment:
nit: since we only care about one possible strategy, I would probably write this with a `guard` rather than a `switch`:
```swift
guard bsonDecoder.options.dateDecodingStrategy == .bsonDateTime else {
    // throw
}
```

Reply:
I had it this way because you can't actually use `==` between two enum types. You can however use `guard case`, which I've updated it to.

Reply:
ohhhh, yeah I guess we would have to explicitly make the type `Equatable` and probably write some custom implementation since it can have associated values. that makes sense.